homeassistant/components/automation/zone.py

```diff
@@ -46,6 +47,23 @@ def zone_automation_listener(entity, from_s, to_s):
             from_match = False
         to_match = condition.zone(hass, zone_state, to_s)
 
+        #
+        #  If zone is "zone.*", trigger automation when entering or leaving
+        #
+        if zone_entity_id == MATCH_ALL and from_s.state != to_s.state:
```

Comment:
Instead of copy pasting the if-statement, integrate it with the other statement. Extract the check into a variable like done for `from_match` and `to_match`.